### ISSUE TITLE
Feature request: tf.nn.dropout noise_shape should support unspecified dimensions

### ISSUE THREAD

**Issue by Faur** [tensorflow/tensorflow#16034]
It would be nice if the noise_shape in [tf.nn.dropout](https://www.tensorflow.org/api_docs/python/tf/nn/dropout) would support unspecified dimensions, and just use the shape of the input tensor, e.g. `-1` or `None`. This way it could be specified as `noise_shape = [-1, 1, 1, -1]` instead of `noise_shape = [k, 1, 1, n]`.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by yongtang** [comment]
Added a PR #16085 for `-1` support.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?
